packages/core/src/card/InteractableCard.tsx

```diff
@@ -50,31 +80,104 @@ export const InteractableCard = forwardRef<
     window: targetWindow,
   });
 
+  const interactableCardGroup = useInteractableCardGroup();
+
+  const interactableCardGroupSelected = interactableCardGroup
+    ? interactableCardGroup.isSelected(value)
+    : selectedProp;
+
+  const disabled = interactableCardGroup?.disabled || disabledProp;
+
+  const [selected, setSelected] = useControlled({
+    controlled: interactableCardGroupSelected,
+    default: Boolean(selectedProp),
+    name: "InteractableCard",
+    state: "selected",
+  });
+
+  const role = interactableCardGroup
+    ? interactableCardGroup.selectionVariant === "multiselect"
+      ? "checkbox"
+      : "radio"
+    : "button";
+
+  const isMultiselect =
+    interactableCardGroup &&
+    interactableCardGroup.selectionVariant === "multiselect";
+
+  const isFirstChild =
+    interactableCardGroup && interactableCardGroup.isFirstChild(value);
+
+  const ariaChecked =
+    role === "radio" || role === "checkbox" ? selected : undefined;
+
+  const handleClick = (event: MouseEvent<HTMLDivElement>) => {
+    if (interactableCardGroup && !disabled) {
+      interactableCardGroup.select(event, value);
+      setSelected(!selected);
+    }
+    onChange?.(event);
+    onClick?.(event);
+  };
+
+  let tabIndex;
+
+  if (interactableCardGroup) {
+    if (disabled) {
+      tabIndex = -1;
+    } else if (isMultiselect) {
+      tabIndex = 0; // All items focusable in multi-select
+    } else {
+      // Single select: Only selected or first item (if none are selected) is focusable
+      tabIndex = selected ? 0 : -1;
+      if (!interactableCardGroup?.value && isFirstChild) {
+        tabIndex = 0;
+      }
+    }
+  } else {
+    tabIndex = disabled ? -1 : 0;
+  }
+
+  const cardRef = useRef<HTMLDivElement>(null);
+  const handleRef = useForkRef(ref, cardRef);
+
+  useLayoutEffect(() => {
```

Comment:
I'm not sure I understand this ? If the card is selected is should take focus, if the car is selected via the keyboard would it not already have focus ?

Reply:
yeah I don't think this is needed, good spot